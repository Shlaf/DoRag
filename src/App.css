
/* Basic styling: */

Body {
    width: 100%;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: RGBA(51, 51, 51, 1.00);
    color: white;

    font-family: 'Roboto', sans-serif;
    font-weight: 100;
}

h1 {
    font-family: 'Roboto', sans-serif;
    font-weight: 100; 
}

.calculator {
    width: 320px;
    text-align: center;
    box-shadow: 0 2px 4px rgba(0,0,0,0.3), 0 2px 4px rgba(0,0,0,0.3);
    padding-top: 5px;
}

.top {
    display: flex;
    justify-content: space-between;
    margin: 0 10px;
}

button.themes {
    background-color: rgba(256,256,256,0);
    font-size: 1em;
    width: 2em;
    height: 2em;
}

.input {
    width: 100%;
    text-align: right;
    font-size: 5em;
    overflow: hidden;
    padding: 0 10px;
}

hr {
    margin: 0;
}

.history {
    padding: 5px 10px;
    text-align: right;
    font-size: 1em;
    opacity: 0.5;
    border-top: 1px rgba(256,256,256,0.5) solid;
}


.button-wrapper {
    display: flex;
    flex-wrap: wrap;
}

button {
    width: 25%;
    height: 80px;
    background-color: rgba(256,256,256,0.1);
    border-top: 1px RGBA(256, 256, 256, 0) solid;
    border-right: 1px RGBA(256, 256, 256, 0) solid;
    border-bottom: 0;
    border-left: 0;
    background-clip: padding-box;
    color: RGBA(256, 256, 256, 0.9);
    font-family: 'Roboto', sans-serif;
    font-weight: 100;
    font-size: 1.7em;
    transition: all 0.2s ease-in-out;
}

button.selected {
    background-color: rgba(256,256,256,0.3);    
}

button:hover {
    background-color: rgba(256,256,256,0.2);
    color: RGBA(256, 256, 256, 1);
    
}

.zero {
    flex-grow: 2;
}