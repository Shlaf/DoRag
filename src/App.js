import React, { Component } from 'react';
import addition from "./functions/addition";
import subtract from "./functions/subtract";
import multiply from "./functions/multiply";
import divide from "./functions/divide";
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <h2>React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}
export default App;
