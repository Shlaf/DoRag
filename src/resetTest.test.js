import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';



describe("clear", ()=>{
  it('handles clear button', () => {
    const component  = renderer.create(<App />);
    component.getInstance().handleInputClick(100);
    let clear = component.toJSON();
    expect(clear).toMatchSnapshot();

    component.getInstance().handelClear();
    clear = component.toJSON();    
    expect(clear).toMatchSnapshot();
  })
})
