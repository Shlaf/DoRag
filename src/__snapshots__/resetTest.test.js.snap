exports[`clear handles clear button 1`] = `
<main>
  <div
    className="calculator red">
    <div
      className="top">
      <button
        className="themes"
        onClick={[Function]}>
        ∞
      </button>
      <div />
      <h1>
        calc
      </h1>
    </div>
    <div
      className="input ">
      <span>
        100
      </span>
    </div>
    <div
      className="button-wrapper">
      <button
        className={null}
        onClick={[Function]}>
        C
      </button>
      <button
        className={null}
        onClick={[Function]}>
        (-)
      </button>
      <button
        className={null}
        onClick={[Function]}>
        <
      </button>
      <button
        className={null}
        onClick={[Function]}>
        /
      </button>
      <button
        className={null}
        onClick={[Function]}>
        1
      </button>
      <button
        className={null}
        onClick={[Function]}>
        2
      </button>
      <button
        className={null}
        onClick={[Function]}>
        3
      </button>
      <button
        className={null}
        onClick={[Function]}>
        *
      </button>
      <button
        className={null}
        onClick={[Function]}>
        4
      </button>
      <button
        className={null}
        onClick={[Function]}>
        5
      </button>
      <button
        className={null}
        onClick={[Function]}>
        6
      </button>
      <button
        className={null}
        onClick={[Function]}>
        -
      </button>
      <button
        className={null}
        onClick={[Function]}>
        7
      </button>
      <button
        className={null}
        onClick={[Function]}>
        8
      </button>
      <button
        className={null}
        onClick={[Function]}>
        9
      </button>
      <button
        className={null}
        onClick={[Function]}>
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}>
        0
      </button>
      <button
        className={null}
        onClick={[Function]}>
        .
      </button>
      <button
        className={null}
        onClick={[Function]}>
        =
      </button>
    </div>
  </div>
</main>
`;

exports[`clear handles clear button 2`] = `
<main>
  <div
    className="calculator red">
    <div
      className="top">
      <button
        className="themes"
        onClick={[Function]}>
        ∞
      </button>
      <div />
      <h1>
        calc
      </h1>
    </div>
    <div
      className="input ">
      <span>
        0
      </span>
    </div>
    <div
      className="button-wrapper">
      <button
        className={null}
        onClick={[Function]}>
        C
      </button>
      <button
        className={null}
        onClick={[Function]}>
        (-)
      </button>
      <button
        className={null}
        onClick={[Function]}>
        <
      </button>
      <button
        className={null}
        onClick={[Function]}>
        /
      </button>
      <button
        className={null}
        onClick={[Function]}>
        1
      </button>
      <button
        className={null}
        onClick={[Function]}>
        2
      </button>
      <button
        className={null}
        onClick={[Function]}>
        3
      </button>
      <button
        className={null}
        onClick={[Function]}>
        *
      </button>
      <button
        className={null}
        onClick={[Function]}>
        4
      </button>
      <button
        className={null}
        onClick={[Function]}>
        5
      </button>
      <button
        className={null}
        onClick={[Function]}>
        6
      </button>
      <button
        className={null}
        onClick={[Function]}>
        -
      </button>
      <button
        className={null}
        onClick={[Function]}>
        7
      </button>
      <button
        className={null}
        onClick={[Function]}>
        8
      </button>
      <button
        className={null}
        onClick={[Function]}>
        9
      </button>
      <button
        className={null}
        onClick={[Function]}>
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}>
        0
      </button>
      <button
        className={null}
        onClick={[Function]}>
        .
      </button>
      <button
        className={null}
        onClick={[Function]}>
        =
      </button>
    </div>
  </div>
</main>
`;
